### PR TITLE
Simplify #doBrowseImplementors and #doBrowseSenders

### PR DESCRIPTION
**Pull request by MarcusDenker**
#doBrowseImplementors and #doBrowseSenders where a bit hard to follow with the odd #browseSelectedSelectorIfGlobal:ifNotGlobal: method.

This PR intead inlines the logic, which makes it much clearer what happenes

### CODE REVIEW

**Review by estebanlm — APPROVED** [review]
not sure how this work as a simplification, but ok :)

**Comment by estebanlm** [comment]
but there are a lot of tests failng

**Comment by MarcusDenker** [comment]
For failing tests, I added https://github.com/pharo-spec/Spec/issues/1415
(they are not a result of this PR)
